### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by bsahil29** [sangwoomo/instagan#20]
I am trying to train the InstaGAN on a smaller dataset of around 40 images from both domains. But I am getting the following error :
**File "/network/home/bansalsa/instagan/data/unaligned_seg_dataset.py", line 49, in read_segs
    segs.append(-torch.ones(segs[0].size()))
IndexError: list index out of range**

I have read the already available issues and double-checked that there is a mask for every image in the trainA and trainB folders but still I'm getting the same error. Also, the images are readable via PIL, I have checked that too. 

This is the screenshot to the output that I'm getting :

![Screenshot 2019-06-04 at 16 43 32](https://user-images.githubusercontent.com/36884804/58912308-c019e280-86e7-11e9-8288-161b0c489542.png)

What could be a reason for getting this error?

**Comment by sangwoomo** [comment]
Hi, it seems that the code fails to read the mask. You'd better to check for which image/mask the code fails to read exactly, e.g., by printing the ID of the image before the failure.
https://github.com/sangwoomo/instagan/issues/5#issuecomment-452587441

**Comment by bsahil29** [comment]
Yeah it worked. There was an issue with reading some of the masks. Thanks! :)

**Comment by Jessicall** [comment]
@bsahil29  hi! I have met the same question and could you please tell me how to solve it?
Thanks a lot!

**Comment by kalai2033** [comment]
@bsahil29 Im getting same error. How did you fix it?

**Comment by kalai2033** [comment]
Hi @sangwoomo  and @bsahil29  I face the same issue...

I have printed the seg_paths ... Every time it breaks at random point. But i have corresponding mask images in both the train and train_seg folder with same name and same dimensions. I cant understand why it fails.

Can you please help me fix it?

**Comment by sangwoomo** [comment]
I'm sorry but I have no idea since I had never encountered this problem.
Can you simply remove the problematic index?

**Comment by MaiKawauchi** [comment]
Hello,  I am trying to train the InstaGAN on the jeans2skirt_ccp. I think the ID of the image is correct. But, this is the screenshot to the output that I'm getting.
What is the cause?
<img width="738" alt="issue" src="https://user-images.githubusercontent.com/49463452/90470806-65876d80-e157-11ea-98cd-0f3f03eee5f7.png">

**Comment by jdunkelheit** [comment]
In addition to missing masks, this issue can also happen when the masks have a wrong name. The mask cannot simply have the exact same name as the image.
Make sure the name of the masks is composed of "image name" + "_0". For example:

>         ├── trainA
>             ├── 1.png
>             ├── 2.png
>             └── ...
>         ├── trainA_seg
>             ├── 1_0.png
>             ├── 2_0.png
>             └── ...
>         ├── trainB
>             ├── 1.png
>             ├── 2.png
>             └── ...
>         ├── trainB_seg
>             ├── 1_0.png
>             ├── 2_0.png
>             └── ...

In case you have more than one mask for the same image, it will be like:
>         ├── trainA
>             ├── 1.png
>             ├── 2.png
>             └── ...
>         ├── trainA_seg
>             ├── 1_0.png
>             ├── 1_1.png
>             ├── 2_0.png
>             ├── 2_1.png
>             └── ...
>         ├── trainB
>             ├── 1.png
>             ├── 2.png
>             └── ...
>         ├── trainB_seg
>             ├── 1_0.png
>             ├── 1_1.png
>             ├── 2_0.png
>             ├── 2_1.png
>             └── ...
